tests/integration/select2-methods.js

```diff
@@ -137,3 +137,70 @@ test('multiple value matches the jquery value', function (assert) {
     'The values should match the jquery values'
   );
 });
+
+test('multiple selection and clearing of grouped options', function (assert) {
+  var container = new MockContainer();
+  var $container = $('<div></div>');
+  var $select = $('<select></select>');
+  $container.append($select);
+
+  var data = [{
+    text: 'Group 1',
+    children: [{
+      id: 1,
+      text: 'Option 1.1'
+    }, {
+      id: 2,
+      text: 'Option 1.2'
+    }]
+  }, {
+    text: 'Group 2',
+    children: [{
+      id: 3,
+      text: 'Option 2.1'
+    }, {
+      id: 4,
+      text: 'Option 2.2'
+    }]
+  }];
+
+  var select = new Select2($select, {
+    multiple: true,
+    data: data
+  });
+  select.render();
+  select.selection.bind(container, $container);
+
+  $select.val(['3', '1']);
+  $select.trigger('change');
+
+  assert.equal(
+    $select.find(':selected').length,
+    2,
+    'Two items should be selected'
+  );
+
+  var $remove = null;
+
+  // Remove the first item
+  $remove = $container.find('.select2-selection__choice__remove');
+  $remove.trigger('click');
+
+  // Open the dropdown menu
+  $container.find('.select2-selection').trigger('click');
+  var $selections = $('.select2-results__option[aria-selected=true]');
```

Comment:
I'm not sure that this `jquery` query (that starts from the global '$' object -- instead of `$container` or anything local to this function) is returning the expected objects.

What I'd like it to find is the selected items in the dropdown for the multi-select created within this test case.

I have a sense that it's retrieving elements from another, unrelated integration test.